cpp/oneapi/dal/data/undirected_adjacency_array_graph.cpp

```diff
@@ -71,58 +58,38 @@ auto get_edge_count_impl(const G &g) noexcept -> edge_size_type<G> {
     return layout->_edge_count;
 }
 
-template edge_size_type<graph64> get_edge_count_impl(const graph64 &g);
 template edge_size_type<graph32> get_edge_count_impl(const graph32 &g);
 
 template <typename G>
 auto get_vertex_degree_impl(const G &g, const vertex_type<G> &vertex) -> vertex_edge_size_type<G> {
     const auto &layout = detail::get_impl(g);
-    return layout->_degrees[vertex];
+    return layout->_vertexes[vertex + 1] - layout->_vertexes[vertex];
 }
 
-template vertex_edge_size_type<graph64> get_vertex_degree_impl(const graph64 &g,
-                                                               const vertex_type<graph64> &vertex);
 template vertex_edge_size_type<graph32> get_vertex_degree_impl(const graph32 &g,
                                                                const vertex_type<graph32> &vertex);
 
 template <typename G>
 auto get_vertex_neighbors_impl(const G &g, const vertex_type<G> &vertex)
     -> const_vertex_edge_range_type<G> {
     const auto &layout = detail::get_impl(g);
-    const_vertex_edge_iterator_type<G> vertex_neighbors_begin(
-        &layout->_edge_offsets[layout->_vertex_neighbors[vertex]]);
-    const_vertex_edge_iterator_type<G> vertex_neighbors_end(
-        &layout->_edge_offsets[layout->_vertex_neighbors[vertex + 1]]);
+    const_vertex_edge_iterator_type<G> vertex_neighbors_begin =
+        layout->_edges.begin() + layout->_vertexes[vertex];
+    const_vertex_edge_iterator_type<G> vertex_neighbors_end =
+        layout->_edges.begin() + layout->_vertexes[vertex + 1];
     auto neighbors_range = std::make_pair(vertex_neighbors_begin, vertex_neighbors_end);
     return neighbors_range;
 }
 
-template const_vertex_edge_range_type<graph64> get_vertex_neighbors_impl(
-    const graph64 &g,
-    const vertex_type<graph64> &vertex);
-
 template const_vertex_edge_range_type<graph32> get_vertex_neighbors_impl(
     const graph32 &g,
     const vertex_type<graph32> &vertex);
 
-template <typename G>
-auto get_vertex_value_impl(const G &g, const vertex_type<G> &vertex) -> vertex_user_value_type<G> {
-    const auto &layout = detail::get_impl(g);
-    return layout->_vertex_value[vertex];
-}
-
-template vertex_user_value_type<graph64> get_vertex_value_impl(const graph64 &g,
-                                                               const vertex_type<graph64> &vertex);
-
-template vertex_user_value_type<graph32> get_vertex_value_impl(const graph32 &g,
-                                                               const vertex_type<graph32> &vertex);
-
 template <typename G>
 void convert_to_csr_impl(const edge_list<vertex_type<G>> &edges, G &g) {
```

Comment:
minor on the naming: such split into convert_to_csr and convert_to_csr_impl does not make big sense - convert_to_csr is implementation itself